### PR TITLE
Clean and tidy code

### PR DESCRIPTION
**Pull request by retrofox**
This PR makes minor corrections in some files that I've had to work. I shoudn't affect the operation of the app.

* eslint fixes
* es6ify 
* @jsdoc improvements
* clean and tidy

### CODE REVIEW

**Comment by mtias** [comment]
@retrofox I think it'd be better to do this in smaller batches, and not across unrelated files, because it's harder to test there are no regressions.

cc @ehg @mcsf @ockham @seear for the menus part

File: client/lib/menu-data/menu-data.js [hunk 1]
```diff
@@ -870,7 +876,9 @@ MenuData.prototype.fetchDefaultMenu = function() {
 	this.data.defaultMenu = false;
 	this.fetchingDefaultMenu = true;
 
-	wpcom.site( requestedSiteID ).postsList( params, function( error, data ) {
+	wpcom
+	.site( requestedSiteID )
```
Comment (mtias) [inline client/lib/menu-data/menu-data.js]: I'm not sure this is more readable this way, to be honest.

**Comment by retrofox** [comment]
> @retrofox I think it'd be better to do this in smaller batches, and not across unrelated files, because it's harder to test there are no regressions.

:+1: Let me split it in a few PRs.

File: client/lib/menu-data/menu-data.js (hunk 1 above)
Comment (nylen) [inline client/lib/menu-data/menu-data.js]: I think it is.  Also, long lines should be avoided.

I'd prefer to indent the subsequent lines:

```
wpcom
    .site( .. )
    .postsList(
        ..
    );
```

but we use the un-indented style in lots of places.

**Comment by nylen** [comment]
+1 to smaller PRs.  Also testing instructions are very helpful - I'm in favor of cleanup changes but I'm much more likely to be able to review them if there are clear instructions to test.  For a lot of changes here that could be as simple as "run the following test suite which covers this code" or "verify that X widget in Calypso still does Y when you load page Z".

**Pull request closed without merging**